test/e2e/util/util.go

```diff
@@ -135,8 +135,19 @@ func WaitForGuestClient(t *testing.T, ctx context.Context, client crclient.Clien
 	return guestClient
 }
 
-func WaitForNReadyNodes(t *testing.T, ctx context.Context, client crclient.Client, n int32, platform hyperv1.PlatformType) []corev1.Node {
+func WaitForNReadyNodes(t *testing.T, ctx context.Context, client crclient.Client, n int32, platform hyperv1.PlatformType, nodePools ...string) []corev1.Node {
 	g := NewWithT(t)
+	var filterSet bool
+	var nodePoolName string
+	// nodePools is a slice of strings which contains a number of NodePools to filter by.
+	if len(nodePools) > 0 {
+		filterSet = true
+		if len(nodePools) > 1 {
+			t.Log("WARNING: We only support 1 object in the nodePools varaidic parameter")
```

Comment:
why are you allowing more in the signature them?

Reply:
I'm assigning the index 0 in the line 148

Reply:
Answered in the next comment :)